src/tables/colr.rs

```diff
@@ -47,16 +189,520 @@ impl FromData for LayerRecord {
+        color.apply_alpha(info.alpha.apply_float_delta(deltas[1]));
+
+        Some(ColorStop { stop_offset, color })
+    }
+}
+
+#[cfg(feature = "variable-fonts")]
+#[derive(Clone)]
+struct VarColorLine<'a> {
+    extend: GradientExtend,
+    colors: LazyArray16<'a, VarColorStopRaw>,
+    palettes: cpal::Table<'a>,
+    foreground_color: RgbaColor,
+}
+
+#[derive(Clone)]
+enum ColorLine<'a> {
+    #[cfg(feature = "variable-fonts")]
+    VarColorLine(VarColorLine<'a>),
+    NonVarColorLine(NonVarColorLine<'a>),
+}
+
+/// A [gradient extend](
+/// https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug)]
+pub struct ColorStop {
+    /// The offset of the color stop.
+    pub stop_offset: f32,
+    /// The color of the color stop.
+    pub color: RgbaColor,
+}
+
+/// A [linear gradient](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#formats-4-and-5-paintlineargradient-paintvarlineargradient)
+#[derive(Clone)]
+pub struct LinearGradient<'a> {
+    /// The `x0` value of the linear gradient.
+    pub x0: f32,
+    /// The `y0` value of the linear gradient.
+    pub y0: f32,
+    /// The `x1` value of the linear gradient.
+    pub x1: f32,
+    /// The `y1` value of the linear gradient.
+    pub y1: f32,
+    /// The `x2` value of the linear gradient.
+    pub x2: f32,
+    /// The `y2` value of the linear gradient.
+    pub y2: f32,
+    /// The extend of the gradient.
+    pub extend: GradientExtend,
+    #[cfg(feature = "variable-fonts")]
+    variation_data: VariationData<'a>,
+    color_line: ColorLine<'a>,
+}
+
+impl<'a> core::fmt::Debug for LinearGradient<'a> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("LinearGradient")
+            .field("x0", &self.x0)
+            .field("y0", &self.y0)
+            .field("x1", &self.x1)
+            .field("y1", &self.y1)
+            .field("x2", &self.x2)
+            .field("y2", &self.y2)
+            .field("extend", &self.extend)
+            // .field("stops", &self.stops(0))
+            .finish()
+    }
+}
+
+impl<'a> LinearGradient<'a> {
+    /// Returns an iterator over the stops of the linear gradient.
+    pub fn stops<'b>(
+        &'b self,
+        palette: u16,
+        #[cfg(feature = "variable-fonts")] coords: &'b [NormalizedCoordinate],
+    ) -> GradientStopsIter<'a, 'b> {
+        GradientStopsIter {
+            color_line: &self.color_line,
+            palette,
+            index: 0,
+            #[cfg(feature = "variable-fonts")]
+            variation_data: self.variation_data,
+            #[cfg(feature = "variable-fonts")]
+            coords,
+        }
+    }
+}
+
+/// A [radial gradient](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#formats-6-and-7-paintradialgradient-paintvarradialgradient)
+#[derive(Clone)]
+pub struct RadialGradient<'a> {
+    /// The `x0` value of the radial gradient.
```

Comment:
I think `of the radial gradient` could be removed.  It's unnecessary.

Reply:
Makes sense. I'll also remove it for the others.